modules/cache_behavior/variables.tf

```diff
@@ -0,0 +1,135 @@
+variable "path_pattern" {
+  description = "The pattern (for example \"images/*.jpg\") that specifies which requests you want this cache behavior to apply to. Not required for the default cache behavior!"
+  type        = string
+  default     = null
+}
+
+variable "allowed_methods" {
+  description = "Controls which HTTP methods CloudFront processes and forwards to your Amazon S3 bucket."
+  type        = list(string)
+  default     = ["GET", "HEAD", "OPTIONS"]
+
+  validation {
+    condition = alltrue([
+      for v in var.allowed_methods : contains(["GET", "HEAD", "POST", "PUT", "DELETE", "CONNECT", "OPTIONS", "TRACE", "PATCH"], v)
+    ])
+    error_message = "The allowed_methods value must be a list of valid HTTP methods."
+  }
+}
+
+variable "cached_methods" {
+  description = "Controls whether CloudFront caches the response to requests using the specified HTTP methods."
+  type        = list(string)
+  default     = ["GET", "HEAD"]
+
+  validation {
+    condition = alltrue([
+      for v in var.cached_methods : contains(["GET", "HEAD", "POST", "PUT", "DELETE", "CONNECT", "OPTIONS", "TRACE", "PATCH"], v)
+    ])
+    error_message = "The cached_methods value must be a list of valid HTTP methods."
+  }
+}
+
+variable "viewer_protocol_policy" {
+  description = "Use this element to specify the protocol that users can use to access the files in the origin when a requests matches this "
+  type        = string
+  default     = "redirect-to-https"
+  validation {
+    condition     = contains(["allow-all", "https-only", "redirect-to-https"], var.viewer_protocol_policy)
+    error_message = "The viewer_protocol_policy value must be one of \"allow-all\", \"https-only\", or \"redirect-to-https\"."
+  }
+}
+
+variable "compress" {
+  description = "Whether you want CloudFront to automatically compress content for web requests that include \"Accept-Encoding: gzip\" in the request header."
+  type        = bool
+  default     = false
+}
+
+variable "min_ttl" {
+  description = "The minimum amount of time (in seconds) that you want objects to stay in CloudFront caches before CloudFront queries your origin to see whether the object has been updated."
+  type        = number
+  default     = 0
```

Comment:
I wonder if this should be 1 so that people have to set it to 0 intentionally in order to enable a browser's `no-cache` behavior from a `<meta>` tag

Reply:
I also wondered that. That is the default for the optimized caching policy in the console. I left it at 0 to keep in line with the old values. I'm for updating it though. Curious what @jduv thinks about this.

Reply:
set to 1